lib/outside/off/off_shops_manager.dart

```diff
@@ -73,7 +78,11 @@ class OffShopsManager {
           'User is out of all countries! Coord: $cameraPos, addr: $address');
       return Err(None());
     }
-
+    //when not in enabled countries, we don't load the products from OFF
+    if (!enabledCountryCodes.contains(countryCode)){
+      Log.d('Country $countryCode is not in the enabled countries list $enabledCountryCodes not fetching shops and products from off');
+      return Err(None());
```

Comment:
Returning error from here will cause the `_getCountryCodeImpl` function to be called each time `_countryCodeOp` is used, and each time it will send requests to Open Street Map (because of `CachedOperation`).

Maybe this check should be done rather in `fetchOffShops` and `fetchVeganBarcodesForShops`?
Or you could create a private getter `_countryCode`:
```
Future<Result<String?, None>> get _countryCode async {
  final result = await _countryCodeOp.result;
  if (result.isOk && !enabledCountryCodes.contains(result.unwrap()) {
    return Ok(null);
  }
}
```
... `_countryCode` then would be used everywhere instead of `_countryCodeOp`.

Also notice how I replaced `Result<String, None>` with `Result<String?, None>` - this way other methods of `OffShopsManager` would understand when they should return an empty collection instead of an error.

Reply:
Ok i didn't see that. I thought i would be more generic. But you are right. Let me fix that!